### PR TITLE
New application types (templates) for the CLI

### PR DESCRIPTION
**Pull request by spericas**
New application types (templates) for the CLI. More templates can be added following the same structure and naming convention that is part of the contract in with the CLI's init command.

Signed-off-by: Santiago Pericasgeertsen <santiago.pericasgeertsen@oracle.com>

### CODE REVIEW

File: apptypes/mp/basic/src/main/resources/META-INF/helidon-archetype-resources.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+pom.xml.mustache
```
Comment (romain-grecourt) [inline apptypes/mp/basic/src/main/resources/META-INF/helidon-archetype-resources.txt:1]: Can you remove this file? This should be generated by the plugin.

File: apptypes/mp/basic/pom.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <build>
+        <extensions>
+            <extension>
+                <groupId>io.helidon.build-tools.archetype</groupId>
+                <artifactId>helidon-archetype-maven-plugin</artifactId>
+                <version>${project.version}</version>
```
Comment (romain-grecourt) [inline apptypes/mp/basic/pom.xml:40]: This should not be `${project.version}`. This should be moved in the parent pom as plugin with extension=true and use the build-tools version property.

File: apptypes/se/basic/pom.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        <version>2.0.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>helidon-apptypes-se-basic</artifactId>
+    <name>Helidon Template Application SE basic</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <build>
```
Comment (romain-grecourt) [inline apptypes/se/basic/pom.xml:35]: Same comment as above, this should be moved to the parent pom and use the build-tools version property.

File: apptypes/se/basic/src/main/resources/META-INF/helidon-archetype-resources.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+pom.xml.mustache
```
Comment (romain-grecourt) [inline apptypes/se/basic/src/main/resources/META-INF/helidon-archetype-resources.txt:1]: Same comment as above, this should be moved.

File: apptypes/mp/basic/pom.xml (hunk 2 above)
Comment (spericas) [reply apptypes/mp/basic/pom.xml:40]: Right, not the correct version. Will fix.

**Comment by spericas** [comment]
Fyi @romain-grecourt 

- Relocation under archetypes
- POM cleanup
- There seems to be an issue regarding the generation of `helidon-archetype-resources.txt` by the plugin. Need to investigate it, so files are still there.
- Will create some tests in a follow-up PR

File: archetypes/mp/basic/src/main/resources/src/main/java/__pkg__/GreetResource.java.mustache [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * A simple JAX-RS resource to greet you. Examples:
+ *
+ * Get default greeting message:
+ * curl -X GET http://localhost:8080/greet
+ *
+ * The message is returned as a JSON object.
+ */
+@Path("/greet")
+@RequestScoped
```
Comment (tomas-langer) [inline archetypes/mp/basic/src/main/resources/src/main/java/__pkg__/GreetResource.java.mustache:27]: This annotation is no longer required.

File: archetypes/mp/basic/src/main/resources/src/main/java/__pkg__/GreetResource.java.mustache [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    public GreetResource(@ConfigProperty(name = "app.greeting") String message) {
+        this.message = message;
+    }
+
+    /**
+     * Return a wordly greeting message.
+     *
+     * @return {@link JsonObject}
+     */
+    @SuppressWarnings("checkstyle:designforextension")
```
Comment (tomas-langer) [inline archetypes/mp/basic/src/main/resources/src/main/java/__pkg__/GreetResource.java.mustache:43]: Please remove these `@SuppressWarnings` annotations. They are not relevant.

File: archetypes/se/basic/src/main/resources/pom.xml.mustache [hunk 7]
```diff
@@ -7,36 +6,26 @@
...
     <properties>
-        <mainClass>${package}.Main</mainClass>
+        <mainClass>{{package}}.Main</mainClass>
     </properties>
 
     <dependencies>
         <dependency>
-            <groupId>io.helidon.webserver</groupId>
-            <artifactId>helidon-webserver</artifactId>
+            <groupId>io.helidon.bundles</groupId>
```
Comment (tomas-langer) [inline archetypes/se/basic/src/main/resources/pom.xml.mustache:22]: Please stop using this bundle.
I think it is confusing - it just adds support for JSON-P.

File: archetypes/se/basic/src/main/resources/src/main/java/__pkg__/Main.java.mustache [hunk 8]
```diff
@@ -49,83 +40,58 @@ public final class Main {
...
-#else
-            .routing(createRouting())
-#end
-#if( $jsonSupport.matches("y|yes|true") )
-            .addMediaSupport(JsonpSupport.create())
-#end
-            .build();
+
+        // Get webserver config from the "server" section of application.yaml
+        ServerConfiguration serverConfig =
```
Comment (tomas-langer) [inline archetypes/se/basic/src/main/resources/src/main/java/__pkg__/Main.java.mustache:50]: This is now deprecated. Please switch to new API (as was in old template in master).

File: archetypes/se/basic/src/main/resources/src/main/java/__pkg__/Main.java.mustache [hunk 9]
```diff
@@ -49,83 +40,58 @@ public final class Main {
...
+        GreetService greetService = new GreetService(config);
         HealthSupport health = HealthSupport.builder()
-                .add(HealthChecks.healthChecks())   // Adds a convenient set of checks
+                .addLiveness(HealthChecks.healthChecks())   // Adds a convenient set of checks
                 .build();
-#end
 
         return Routing.builder()
-#if( $healthSupport.matches("y|yes|true") )
+                .register(JsonSupport.create())
```
Comment (tomas-langer) [inline archetypes/se/basic/src/main/resources/src/main/java/__pkg__/Main.java.mustache:89]: This is wrong place to register this.
Please switch to new media support
Should be registered on `Server.builder().addMediaSupport(JsonpSupport.create())`

**Pull request merged**